klippy/extras/force_move.py

```diff
@@ -131,8 +131,16 @@ def cmd_SET_KINEMATIC_POSITION(self, gcmd):
         x = gcmd.get_float('X', curpos[0])
         y = gcmd.get_float('Y', curpos[1])
         z = gcmd.get_float('Z', curpos[2])
-        logging.info("SET_KINEMATIC_POSITION pos=%.3f,%.3f,%.3f", x, y, z)
-        toolhead.set_position([x, y, z, curpos[3]], homing_axes=(0, 1, 2))
+        clear = gcmd.get('CLEAR', '').upper()
+        axes = ['X', 'Y', 'Z']
```

Comment:
this should be based on printer['toolhead'].homed_axes

Reply:
this seems to work

```py
        clear = gcmd.get('CLEAR', '').upper()
        curtime = self.printer.get_reactor().monotonic()
        kin_status = toolhead.get_kinematics().get_status(curtime)
        axes = list(kin_status['homed_axes'].upper())
        clear_axes = [axes.index(a) for a in axes if a in clear]
        homing_axes = tuple(i for i, a in enumerate(['X', 'Y', 'Z']) if a in axes and a not in clear)
```

Reply:
the `clear` variable may need to be sorted alphabetically in case user inputs CLEAR=ZYX

Reply:
Thanks for noticing this, I'll take a look this weekend and make sure it works in those cases as well.

Reply:
Upon closer inspection this is intended behavior, see response [below](https://github.com/Klipper3d/klipper/pull/6262#issuecomment-2548252460).